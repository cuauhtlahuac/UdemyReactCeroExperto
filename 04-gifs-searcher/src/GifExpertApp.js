import React, { useState } from 'react';

import GifGrid from 'components/GifGrid';
import AddCategory from 'components/AddCategory';

const GifExpertApp = () => {
	const initialState = [ 'Dragon ball' ];
	const [ categories, setCategories ] = useState(initialState);
	const [ categoryDictionary, setCategoryDictionary ] = useState({});
	const [ sameValue, setSameValue ] = useState(false);

	if (sameValue) {
		return (
			<div>
				<h1>You wrote it again, please write other category </h1>
				<button onClick={() => setSameValue(false)}>return</button>
			</div>
		);
	}

	return (
		<section>
			<h2 className="gif-title">Gif Searcher</h2>

			<AddCategory
				setCategories={setCategories}
				setCategoryDictionary={setCategoryDictionary}
				setSameValue={setSameValue}
				categoryDictionary={categoryDictionary}
			/>
			<hr />

			{categories.map(category => {
				return <GifGrid category={category} key={category} />;
			})}
		</section>
	);
};

export default GifExpertApp;
