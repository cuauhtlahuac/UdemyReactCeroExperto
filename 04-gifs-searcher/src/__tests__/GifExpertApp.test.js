import React from 'react';
import { shallow } from "enzyme"
import GifExpertApp from "GifExpertApp"

describe('GifExpertApp Tests', () => {

    test('should render correctly', () => {
        const component = shallow( <GifExpertApp /> )
        
        expect( component ).toMatchSnapshot();
        expect( component.find('GifGrid').exists()).toBe( false );
    })
    
    test('should render gifGrid with dragon ball', () => {
        const categories = [ 'Dragon ball', 'One punch man * ' ]
        const component = shallow( <GifExpertApp definedCategories={ categories }/> );
        
        expect( component ).toMatchSnapshot();
        expect( component.find('GifGrid').exists() ).toBe( true );
        expect( component.find('GifGrid').length ).toBe( categories.length );
        expect( component.find('GifGrid').at(0).prop('category') ).toBe( categories[0] );
        expect( component.find('GifGrid').at(1).prop('category') ).toBe( categories[1] );
    })
})
