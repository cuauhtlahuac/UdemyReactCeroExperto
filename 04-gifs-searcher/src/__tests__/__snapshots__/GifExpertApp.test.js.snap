// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp Tests should render correctly 1`] = `
<section>
  <h1
    className="gif-title"
  >
    Gif Searcher
  </h1>
  <AddCategory
    categoryDictionary={Object {}}
    setCategories={[Function]}
    setCategoryDictionary={[Function]}
    setSameValue={[Function]}
  />
  <hr />
</section>
`;

exports[`GifExpertApp Tests should render gifGrid with dragon ball 1`] = `
<section>
  <h1
    className="gif-title"
  >
    Gif Searcher
  </h1>
  <AddCategory
    categoryDictionary={Object {}}
    setCategories={[Function]}
    setCategoryDictionary={[Function]}
    setSameValue={[Function]}
  />
  <hr />
  <GifGrid
    category="Dragon ball"
    key="Dragon ball"
  />
  <GifGrid
    category="One punch man * "
    key="One punch man * "
  />
</section>
`;
